[feature] make ntpd_leapfile optional

trombik opened the pull request:
fixes #24

Commit:
[feature] make ntpd_leapfile optional

fixes #24

Commit:
bump retries

"ntpd sync reaches all servers" failed on default-ubuntu-1404-amd64

Commit:
bump retry again

without-leapfile-ubuntu-1404-amd64 still fails to sync

Commit:
[documentation] document the new behaviour

trombik commented:
@mitsururike  what is the purpose of `ntpd sync reaches all servers` test? the test makes Jenkins build so fragile. just a single failure of  NTP upstream causes the test to fail. no NTP services are supposed to guarantee availability all the time. it looks to me that the test is aimed at either the NTP service used or the `ntpd`, not the role, and depends on a false assumption without giving benefits.

mitsururike commented:
previously, all servers are not reachable because of wrong restrictions and stayed in "INIT" state.
the test should be modified to check that at least a server is not in "INIT" state.

Commit:
[bugfix] do not bailout when any upstream is INIT

the previous example bails out any one upstream is in INIT state, which
means the upstream is not working or reachable. the external resources
used in the test are not supposed to be available all the time. if not
all upstreams are in INIT state, fine.

mitsururike commented on tests/serverspec/default_spec.rb:
```diff
@@ -90,8 +90,11 @@
-  it "reaches all servers" do
-    expect(command("ntpq -pn").stdout).not_to match(/\.INIT\./)
+  it "reaches at least one server in non-INIT state" do
+    refids = command("ntpq -pn | tail -n +3 | ntpq -pn | tail -n +3 | awk '! /\.POOL\./ { print $2 }'").stdout.split("\n")
```
typo: duplicated "ntpq -pn | tail -n +3"

Commit:
remove a duplicate

mitsururike commented on tests/serverspec/without_leapfile_spec.rb:
```diff
@@ -0,0 +1,103 @@
+    expect(command("ntpq -pn").stdout).to match(/\*\d+\.\d+\.\d+\.\d+ /)
+  end
+  it "reaches all servers" do
+    expect(command("ntpq -pn").stdout).not_to match(/\.INIT\./)
```
the test should be modified as in the "default_spec.rb".

mitsururike commented on tests/serverspec/without_leapfile_spec.rb:
```diff
@@ -0,0 +1,103 @@
+end
+
+describe file(leap_file) do
+  it { should_not be_file }
```
is this a typo ? s/be_file/exist/ ?
or it can be a symlink, directory etc.

mitsururike commented on tests/serverspec/without_leapfile_spec.rb:
```diff
@@ -0,0 +1,103 @@
+  its(:stderr) { should eq("") }
+end
+
+describe "ntpd sync", retry: 120, retry_wait: 1 do
```
s/120/30/ as in the "default_spec.yml"

Commit:
sync with default_spec.rb
